extensions/ql-vscode/src/pure/files.ts

```diff
@@ -107,3 +107,14 @@ export async function* walkDirectory(
     }
   }
 }
+
+/**
+ * Error thrown from methods from the 'fs' module.
+ */
+export interface IOError {
```

Comment:
From [the types in the `fs` module](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2928e1eb0097809e91ec7a77a845329340d03898/types/node/v16/fs.d.ts#L1042), it seems like there's already a type for this: `NodeJS.ErrnoException`. I don't think we should use it since we're only checking for the `code` field, but it's interesting to see that the type also contains other fields.

Reply:
Ah I didn't spot that type existed. I think that's what we're getting in practice though. When I tried logging one of these exceptions I saw all of those fields.

I'm not sure if we should use that type of not. I think in practice the exception will be of that type, but as you say we currently don't really want all of the fields. So in practice it would just mean modelling unneccessary stuff and make the `isErrnoException` method we'd have to write be more complex.

For now I'll leave it as this custom type with just the `code` field, but I'll make a note about the `NodeJS.ErrnoException` type so anyone who reads this code in the future knows it exists.